source/extensions/tracers/opentelemetry/resource_detectors/static/static_config_resource_detector.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "envoy/extensions/tracers/opentelemetry/resource_detectors/v3/static_config_resource_detector.pb.h"
+#include "envoy/server/factory_context.h"
+
+#include "source/common/common/logger.h"
+#include "source/extensions/tracers/opentelemetry/resource_detectors/resource_detector.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace OpenTelemetry {
+
+/**
+ * @brief A resource detector that extracts attributes from static config
+ *
+ */
+class StaticConfigResourceDetector : public ResourceDetector,
+                                     Logger::Loggable<Logger::Id::tracing> {
+public:
+  StaticConfigResourceDetector(const envoy::extensions::tracers::opentelemetry::resource_detectors::
+                                   v3::StaticConfigResourceDetectorConfig& config,
+                               Server::Configuration::TracerFactoryContext&)
+      : config_(config) {}
+  Resource detect() override;
+
+private:
+  const envoy::extensions::tracers::opentelemetry::resource_detectors::v3::
+      StaticConfigResourceDetectorConfig config_;
```

Comment:
Rather than storing the whole proto config here, I think you could just to store the attributes by flat_hash_map